Assets/Shaders/Caustics.gdshader

```diff
@@ -1,43 +1,42 @@
 shader_type spatial;
-render_mode blend_add, cull_front, depth_test_disable, ambient_light_disabled;
+// Additive blending to mix multiple lights.
+// Front face culling so we can be inside the bounding box.
+// Disable depth testing so the caustics always draw.
+// Disable ambient lighting, so the ambient light does not affect the color twice.
+render_mode blend_add, cull_front, depth_test_disabled, ambient_light_disabled;
 
-uniform sampler2D caustics_noise;
+uniform sampler2D caustics_noise: filter_linear;
 
-uniform vec3 box_size = vec3(1.0);
+uniform vec3 box_size = vec3(4.0);
 
 // Caustics Parameters
-uniform float caustics_scale = 1.0;
+uniform float caustics_scale = 4.0;
 uniform float caustics_speed = 0.1;
-uniform float caustics_strength = 0.5;
-uniform float caustics_power = 1.0;
-uniform float caustics_fade_radius = 0.8;
-uniform float caustics_fade_strength = 0.5;
-uniform float caustics_luminance_mask_strength = 1.2;
-uniform float chroma_split = 0.1;
-uniform sampler2D luma_gradient : hint_white;
+uniform float caustics_strength: hint_range(0.0, 4.0) = 1.0;
+uniform float caustics_power: hint_range(0.25, 8.0) = 2.0;
+uniform float caustics_fade_radius = 2.0;
+uniform float caustics_fade_strength: hint_range(0.0,1.0) = 0.5;
+uniform float caustics_luminance_mask_strength: hint_range(-2.0, 2.0) = 0.2;
+uniform float chroma_split: hint_range(0.0, .02) = 0.002;
+uniform sampler2D luma_gradient : hint_default_white;
 
-varying mat4 vertex_camera;
-varying mat4 vertex_world;
-varying mat4 vertex_projection;
-varying mat4 vertex_modelview;
-void vertex(){
-	vertex_camera = CAMERA_MATRIX;
-	vertex_world = WORLD_MATRIX;
-	vertex_projection = PROJECTION_MATRIX;
-	vertex_modelview = MODELVIEW_MATRIX;
-}
+// Screen Textures
+uniform sampler2D DEPTH_TEXTURE: source_color, hint_depth_texture;
+uniform sampler2D SCREEN_TEXTURE: source_color, hint_screen_texture;
+
+void vertex(){}
 
 float insideBox3D(vec3 v, vec3 bottomLeft, vec3 topRight) {
 	vec3 s = step(bottomLeft, v) - step(topRight, v);
 	return s.x * s.y * s.z; 
 }
 
-float check_encompass_point(vec3 p){
+float check_encompass_point(vec3 p, mat4 world_matrix){
```

Comment:
I think it's cleaner to pass in matrices than use varyings, and this brings the number of vertex->fragment varyings to 0.